spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/kafka/KafkaAutoConfiguration.java

```diff
@@ -138,4 +145,38 @@ public KafkaAdmin kafkaAdmin() {
 		return kafkaAdmin;
 	}
 
+	@Configuration
+	@ConditionalOnClass(StreamsBuilder.class)
+	public static class KafkaStreamsConfiguration {
+
+		@Bean(KafkaStreamsDefaultConfiguration.DEFAULT_STREAMS_CONFIG_BEAN_NAME)
+		public Properties defaultKafkaStreamsConfig(KafkaProperties properties) {
+			Properties streamsConfig = new Properties();
+			properties.buildStreamsProperties().forEach((k, v) -> {
+				String value = v.toString();
+				if (v instanceof List && value.length() > 1) {
+					// trim [...] - revert to comma-delimited list
+					value = value.substring(1, value.length() - 1);
+				}
+				streamsConfig.put(k, value);
+			});
+			return streamsConfig;
+		}
+
+		@Bean
+		public Object kafkaStreamsFactoryBeanConfigurer(
```

Comment:
`Object` ?

Reply:
I used a `NullBean` (return null) to avoid creating an explicit `...Configurer` class just for this one property; I can do that if you prefer.

Reply:
I am not I understand what that means but I'd rather have something that backs off with a property type if we should do something.